xapian-letor/tests/api_letor.cc

```diff
@@ -1414,3 +1414,80 @@ DEFINE_TESTCASE(createfeaturevector_tfdoclencolllfcolllen, generated)
 
     return true;
 }
+namespace Xapian {
+
+class XAPIAN_VISIBILITY_DEFAULT CustomFeature : public Feature {
+  public:
+    CustomFeature() {
+	need_stat(TERM_FREQUENCY);
+	need_stat(DOCUMENT_LENGTH);
+	need_stat(COLLECTION_TERM_FREQ);
+	need_stat(COLLECTION_LENGTH);
+	need_stat(INVERSE_DOCUMENT_FREQUENCY);
+    }
+    std::vector<double> get_values() const {
+	return vector<double>();
+    }
+    std::string name() const {
+	return "CustomFeature";
+    }
+    void test_stats() {
+	// test for term frequency
+	TEST_EQUAL(get_termfreq("ZStiger"), 1);
+	TEST_EQUAL(this->get_termfreq("ZXDtiger"), 6);
+	TEST_EQUAL(this->get_termfreq("Ztiger"), 2);
+
+	// test for inverse document frequency
+	TEST_EQUAL_DOUBLE(get_inverse_doc_freq("ZStiger"), 0.176091259055681);
+	TEST_EQUAL_DOUBLE(get_inverse_doc_freq("ZXDtiger"), 0.176091259055681);
+	TEST_EQUAL_DOUBLE(get_inverse_doc_freq("Ztiger"), 0.176091259055681);
+
+	// test for document length
+	TEST_EQUAL(get_doc_length("title"), 4);
+	TEST_EQUAL(get_doc_length("body"), 182);
+	TEST_EQUAL(get_doc_length("whole"), 186);
+
+	// test for collection length
+	TEST_EQUAL(get_collection_length("title"), 13);
+	TEST_EQUAL(get_collection_length("body"), 509);
+	TEST_EQUAL(get_collection_length("whole"), 522);
+
+	// test for collection term frequency
+	TEST_EQUAL(get_collection_termfreq("ZStiger"), 1);
+	TEST_EQUAL(get_collection_termfreq("ZXDtiger"), 6);
+	TEST_EQUAL(get_collection_termfreq("Ztiger"), 2);
+    }
+};
+
+}
+
+DEFINE_TESTCASE(populatefeature, generated) {
+    vector<Xapian::Feature*> f;
+    Xapian::CustomFeature* f1 = new Xapian::CustomFeature();
+    f.push_back(f1);
```

Comment:
I'd suggest just using `f.push_back(new Xapian::CustomFeature());` - it's if anything slightly clearer than using the temporary variable, and avoids the potentially confusing presence of variables `f1` and `fl` in the same function (they look quite similar in some fonts).

Reply:
We can't do that since I am using it to call test_stats() below but I have changed the name so it doesn't sound confusing anymore.